### ISSUE TITLE
The WebView is crashing on iOS In new architecture.

### ISSUE THREAD

**Issue by nikunj95749** [react-native-webview/react-native-webview#3353]
**Bug description:**

**To Reproduce:**

**Expected behavior:**

**Screenshots/Videos:**
![C60C9606-9013-423A-8A11-BB4122A92758](https://github.com/react-native-webview/react-native-webview/assets/48313639/b041bf28-409e-4e6d-ba9e-37b669f63c85)

**Environment:**
 - OS: Sonoma
 - OS version: 14.3.1
 - react-native version: "0.72.5",
 - react-native-webview version: "13.6.4"
 
 error => 
 0x1103fcc40 - [PID=6541] WebProcessProxy::didClose: (web process 0 crash)
0x1103fcc40 - [PID=6541] WebProcessProxy::processDidTerminateOrFailedToLaunch: reason=Crash
Error acquiring assertion: <Error Domain=RBSServiceErrorDomain Code=1 "target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit" UserInfo={NSLocalizedFailureReason=target is not running or doesn't have entitlement com.apple.runningboard.assertions.webkit}>
0x110053000 - ProcessAssertion::acquireSync Failed to acquire RBS assertion 'XPCConnectionTerminationWatchdog' for process with PID=6541, error: (null)
0x11e20e418 - [pageProxyID=37, webPageID=38, PID=6541] WebPageProxy::processDidTerminate: (pid 6541), reason=Crash
0x11e20e418 - [pageProxyID=37, webPageID=38, PID=6541] WebPageProxy::dispatchProcessDidTerminate: reason=Crash

**Comment by ywisax** [comment]
Same problem in react-native-macos

**Comment by gh-pro** [comment]
Hi, same probleme

**Comment by iqra-khaliq** [comment]
Hi, were you able to resolve this issue?

**Comment by nikunj95749** [comment]
Here is the link open this issue again => https://github.com/react-native-webview/react-native-webview/issues/3702

**Comment by heidiv** [comment]
I was having this issue as well - here's how I fixed it:

My ios app has a bunch of webviews and everything initially worked great but with use (updating, swiping, adding, etc) it would hit low memory and hang until eventually crashing/being "killed by the operating system because it is using too much memory". I used xcode Instruments (product -> profile -> leaker) to observe this while testing on my device.

I tried using onContentProcessDidTerminate on the webview to reload webviews when Web Process Terminated happened but the memory issue/crashing remained.

What I had to do was refactor my code to use a state variable for conditional rendering and the memory settled down/app stopped crashing!

---
Conditionally render the WebView component. When it's no longer needed, set the condition to false, which will remove the component from the view hierarchy and allow it to be garbage collected.

    {isVisible && (
      <WebView
        source={{ uri: 'your_url' }}
        // other props
      />
    )}
